db/schema_tables.cc

```diff
@@ -1095,26 +1166,88 @@ future<> store_column_mapping(distributed<service::storage_proxy>& proxy, schema
 // and or filesystem calls, e.g. fsync.
 constexpr size_t max_concurrent = 8;
 
+// Extracts the names of tables affected by a schema mutation.
+// The mutation must target one of the tables in schema_tables_holding_schema_mutations().
+static
+table_selector get_affected_tables(const sstring& keyspace_name, const mutation& m) {
+    const schema& s = *m.schema();
+    auto get_table_name = [&] (const clustering_key& ck) {
+        // The first component of the clustering key in each table listed in
+        // schema_tables_holding_schema_mutations contains the table name.
```

Comment:
One would have to add this new table to `schema_tables_holding_schema_mutations` explicitly, so it would have to involve an intentional mistake on his part.

You want to have a unit test which walks over tables `schema_tables_holding_schema_mutations` and checks if the clustering column is called "table_name"?

Reply:
Adding to schema_tables_holding_schema_mutations doesn't mean the adder knows there must be a clustering key called table_name. So let's check it.